### PR TITLE
Upgrade to Babel 6

### PR DESCRIPTION
**Pull request by taion**
It turns out what I was doing with the output of `matchPattern` was an abuse and only worked in loose mode. Granted, we're still using loose mode, but we might as well not abuse ES6 semantics.

### CODE REVIEW

**Comment by taion** [comment]
It turns out what I was doing with the output of `matchPattern` was an abuse and only worked in loose mode. Granted, we're still using loose mode, but we might as well not abuse ES6 semantics.

File: modules/PatternUtils.js [hunk 1]
```diff
@@ -94,37 +94,29 @@ export function matchPattern(pattern, pathname) {
   }
 
   const match = pathname.match(new RegExp(`^${regexpSource}`, 'i'))
+  if (match == null) {
+    return null
```
Comment (timdorr) [inline modules/PatternUtils.js]: Are we sure we want to change the possible output signature of an exported function?

**Comment by timdorr** [comment]
Maybe this should be against `next`? Babel 5 isn't hurting us right now, so I'd rather deal with the fallout in a major when things are going to change more substantially anyways.

File: modules/PatternUtils.js (hunk 1 above)
Comment (taion) [inline modules/PatternUtils.js]: It's explicitly not part of the public API.

Comment (timdorr) [inline modules/PatternUtils.js]: I mean for internal usage. At least make this into:
```js
return {
  remainingPathname: null,
  paramNames: null,
  paramValues: null
}
```

Comment (taion) [inline modules/PatternUtils.js]: I updated all consumers of `matchPattern`. It's a lot clearer to check for a `null` response value than check for e.g. `matched.paramNames == null`.

**Comment by taion** [comment]
It lets us drop `displayName` from the `withRouter` examples, though. Redux went from Babel 5 to Babel 6 between v3.2.0 to v3.2.1, so there's nothing specifically wrong with Babel 6 upgrades.

**Comment by timdorr** [comment]
Eh, OK. It's your funeral if this doesn't work :P

**Pull request merged**